File: servicejc-backend/src/main/java/com/backend/servicejc/model/UserAddressModel.java
package com.backend.servicejc.model;

/**
 * Modelo para representar la dirección detallada del usuario,
 * que se guarda como un objeto anidado en el documento de Usuario.
 */
public class UserAddressModel {
    
    // Campos que recibes del modelo Dart (Flutter)
    private String provincia;
    private String distrito;
    private String corregimiento;
    private String callePrincipal; // Mapeado desde 'barrio' en Flutter
    private String referencias;    // Mapeado desde 'house' en Flutter
    
    // NUEVOS CAMPOS: Coordenadas del mapa
    private Double latitude;
    private Double longitude;

    // Constructor vacío requerido por Firestore/Jackson
    public UserAddressModel() {}

    // Constructor con todos los campos (incluyendo coordenadas)
    public UserAddressModel(String provincia, String distrito, String corregimiento, 
                            String callePrincipal, String referencias, 
                            Double latitude, Double longitude) {
        this.provincia = provincia;
        this.distrito = distrito;
        this.corregimiento = corregimiento;
        this.callePrincipal = callePrincipal;
        this.referencias = referencias;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    // --- Getters y Setters ---

    public String getProvincia() {
        return provincia;
    }

    public void setProvincia(String provincia) {
        this.provincia = provincia;
    }

    public String getDistrito() {
        return distrito;
    }

    public void setDistrito(String distrito) {
        this.distrito = distrito;
    }

    public String getCorregimiento() {
        return corregimiento;
    }

    public void setCorregimiento(String corregimiento) {
        this.corregimiento = corregimiento;
    }

    public String getCallePrincipal() {
        return callePrincipal;
    }

    public void setCallePrincipal(String callePrincipal) {
        this.callePrincipal = callePrincipal;
    }

    public String getReferencias() {
        return referencias;
    }

    public void setReferencias(String referencias) {
        this.referencias = referencias;
    }

    // Nuevos Getters y Setters para coordenadas
    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }
}

File: servicejc/lib/models/user_address_model.dart
class UserAddressModel {
  final String province;
  final String district;
  final String corregimiento;
  final String barrio; // Se enviará como 'callePrincipal'
  final String house;  // Se enviará como 'referencias'
  
  // 1. AGREGAR PROPIEDADES AQUÍ
  final double? latitude;
  final double? longitude;

  UserAddressModel({
    required this.province,
    required this.district,
    required this.corregimiento,
    required this.barrio,
    required this.house,
    // 2. INICIALIZARLAS CON 'this.'
    this.latitude,
    this.longitude,
  });

  // Método toJson corregido para incluir coordenadas
  Map<String, dynamic> toJson() {
    return {
      'provincia': province,
      'distrito': district,
      'corregimiento': corregimiento,
      'callePrincipal': barrio,
      'referencias': house,
      // 3. AGREGAR AL MAPA JSON
      'latitude': latitude,
      'longitude': longitude,
    };
  }

  factory UserAddressModel.fromJson(Map<String, dynamic> json) {
    return UserAddressModel(
      province: json['provincia'] as String? ?? '',
      district: json['distrito'] as String? ?? '',
      corregimiento: json['corregimiento'] as String? ?? '',
      barrio: json['callePrincipal'] as String? ?? '',
      house: json['referencias'] as String? ?? '',
      // 4. LEER DEL MAPA (Manejo seguro de tipos numéricos)
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
    );
  }
}